app/app/Http/Controllers/HomeController.php

```diff
@@ -23,8 +23,24 @@ public function index(Request $request)
 			'location_tags' => LocationTag::orderBy('name')->get(),
 			'is_authenticated' => BaseUser::isSignedIn(),
 			'address_default' => BaseUser::getDefaultAddress(),
-			'address_value' => $address_value
+			'address_value' => $address_value,
+      'google_map_api_key' => config('app.google_map_api_key'),
+      'turn_off_maps'      => config('app.turn_off_maps')
 			]);
     }
 
-}
\ No newline at end of file
+    public function saveuserlocation(Request $request)
+    {
+          $latitude   = $request->latitude;
+          $longitude = $request->longitude;
+          $address   = $request->address;
+
+          BaseUser::setAddress($address);
+          BaseUser::setLongitude($longitude);
+          BaseUser::setLatitude($latitude);
+
+          echo session("location_search_text");
```

Comment:
Remove the echo statement because it is used only for getting it to work.  Controllers shouldn't write anything to standard output.